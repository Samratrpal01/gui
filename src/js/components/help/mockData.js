export const helpProps = {
  docsVersion: undefined,
  isHosted: false,
  menderVersion: 'master',
  isEnterprise: true,
  tenantCapabilities: {},
  tokens: [],
  versions: {
    repos: {
      mender: '3.3.0',
      'mender-artifact': '3.8.0',
      'mender-binary-delta': '1.3.1',
      'mender-cli': '1.8.0',
      'mender-configure-module': '1.0.4',
      'mender-connect': '2.0.2',
      'mender-convert': '3.0.0',
      'mender-gateway': '1.0.0',
      'monitor-client': '1.2.0'
    },
    releaseDate: '01-01-2003'
  }
};
