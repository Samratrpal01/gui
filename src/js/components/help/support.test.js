import React from 'react';

import { undefineds } from '../../../../tests/mockData';
import { render } from '../../../../tests/setupTests';
import { helpProps } from './mockData';
import Support from './support';

describe('Support Component', () => {
  it('renders correctly', async () => {
    const { baseElement } = render(<Support {...helpProps} />);
    const view = baseElement.firstChild.firstChild;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });
});
