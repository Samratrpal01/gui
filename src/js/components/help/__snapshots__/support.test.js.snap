// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Support Component renders correctly 1`] = `
.emotion-0 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.25rem;
}

<div>
  <h2>
    Contact support
  </h2>
  <p>
    Do you have a paid plan and need to contact support?
  </p>
  <p>
    Send an email to
     
    <a
      href="mailto:support@mender.io"
      rel="noopener noreferrer"
      target="_blank"
    >
      support@mender.io
    </a>
    , including your organization name and how we can help you.
  </p>
  <h4>
    Reporting bugs
  </h4>
  <p>
    To help us best respond to bug reports, please include:
    <ul>
      <li>
        Which software versions you are using
      </li>
      <li>
        If the issue can be reproduced, what are the steps to reproduce it?
      </li>
      <li>
        In case of a UI issue, any screenshots or screen recordings
      </li>
    </ul>
  </p>
  <h3>
    Trial or open source users
  </h3>
  <div>
    <p>
      Find technical help at
    </p>
    <a
      href="https://hub.mender.io"
      rel="noopener noreferrer"
      target="_blank"
    >
      <span>
        <img
          src="mender-hub-logo.png"
          style="width: 50%;"
        />
      </span>
    </a>
    <p>
      Mender Hub is a forum where you can find help, ask technical questions and start discussions with our active and knowledgeable community.
    </p>
    <a
      href="https://hub.mender.io"
      rel="noopener noreferrer"
      target="_blank"
    >
      <span>
        Ask a question at Mender Hub
      </span>
       
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall emotion-0"
        data-testid="LaunchIcon"
        focusable="false"
        style="vertical-align: text-bottom;"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
        />
      </svg>
    </a>
  </div>
</div>
`;
